PyHook.py

```diff
@@ -41,8 +49,11 @@ def wait_for_process(process_name, tag_name, hook_function):
                 running_pids.append(process.pid)
                 print(f"[+] Found {tag_name} Window")
                 hook_function(process.pid)
-            # TODO: elif check if pids still alive -> if not then remove from running_pids
-
+            else:
+                for pid in running_pids:
```

Comment:
This solution will work, but it'll do unnecessary work...
Why get all the running processes in the system instead for querying about the specific ones?
_____
Consider the following change:
 ```python
for pid in running_pids:
    if not psutil.pid_exists(pid):
        running_pids.remove(pid)
```
I suggest to use the above snippet to have a neater/more readable code, but you can also do this cool filter:
```python
running_pids = filter(psutil.pid_exists, running_pids)
```
You know what, I convinced my self, just use the filter... it's much better! :smile:

Reply:
Will do!